src/collision.cpp

```diff
@@ -489,7 +489,7 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		} else if (nearest_dtime > 0) {
 			// updated average speed for the sub-interval up to nearest_dtime
 			aspeed_f = *speed_f + accel_f * 0.5f * nearest_dtime;
-			*pos_f += truncate(aspeed_f * nearest_dtime, 100.0f);
```

Comment:
I am not sure this is related to the immediate issue to be fixed.

Reply:
Right. It's not necessary to fix the slippery node issue. However, I would still prefer to have this code identical to the case above for consistency.
Truncating the position (and related multiplication / division losses) seems like a hacky solution to fix broken collision detection code.